pkg/operator/ceph/csi/spec.go

```diff
@@ -246,6 +247,29 @@ func startDrivers(clientset kubernetes.Interface, rookclientset rookclient.Inter
 	CephFSDriverName = tp.DriverNamePrefix + "cephfs.csi.ceph.com"
 	RBDDriverName = tp.DriverNamePrefix + "rbd.csi.ceph.com"
 
+	if ver.Major == KubeMinMajor && ver.Minor < kubeMinVerForV1csiDriver {
```

Comment:
```suggestion
csiDriverobj = beta1CsiDriver{}
if ver.Major > KubeMinMajor || ver.Major == KubeMinMajor && ver.Minor > kubeMinVerForV1csiDriver {
		// In case of an k8s version upgrade, delete the beta CSIDriver object;
		// before the creation of updated v1 object to avoid conflicts.
		// Ignore if not found.
		if EnableRBD  {
			err = csiDriverobj.deleteCSIDriverInfo(ctx, clientset, RBDDriverName)
			if err != nil {
				logger.Errorf("failed to delete %q Driver Info. %v", RBDDriverName, err)
			}
		}
		if EnableCephFS  {
			err = csiDriverobj.deleteCSIDriverInfo(ctx, clientset, CephFSDriverName)
			if err != nil {
				logger.Errorf("failed to delete %q Driver Info. %v", CephFSDriverName, err)
			}
		}
		csiDriverobj = v1CsiDriver{}
	}
```

Reply:
Thanks, seems like a better approach!